packages/browser-wallet/src/popup/pages/AddTokens/AddTokens.tsx

```diff
@@ -0,0 +1,190 @@
+            {(f) => (
+                <>
+                    <Input
+                        register={f.register}
+                        label={t('contractIndex')}
+                        name="contractIndex"
+                        rules={{
+                            required: t('indexRequired'),
+                        }}
+                    />
+                    <Submit className="tokens__add__submit">{t('chooseContract')}</Submit>
+                </>
+            )}
+        </Form>
+    );
+}
+
+/**
+ * Displays a CIS-2 Token
+ */
+function DisplayToken({ metadata }: { metadata: TokenMetadata }) {
+    return (
+        <div className="tokens__add__element" title={metadata.description}>
+            {metadata.display?.url && (
+                <img alt={metadata.name} className="tokens__add__token-display" src={metadata.display.url} />
+            )}
+            <p>{metadata.name}</p>
+        </div>
+    );
+}
+
+/**
+ * Component used to pick token from a CIS-2 compliant smart contract instance.
+ */
+function PickTokens({
+    contractDetails,
+    onFinish,
+    defaultTokens,
+}: {
+    contractDetails: ContractDetails;
+    onFinish: (updatedTokenList: TokenIdAndMetadata[]) => void;
+    defaultTokens: TokenIdAndMetadata[];
+}) {
+    const { t } = useTranslation('account', { keyPrefix: 'tokens' });
+    const client = useAtomValue(jsonRpcClientAtom);
+    const [accountTokens, setAccountTokens] = useState<TokenIdAndMetadata[]>(defaultTokens);
+    const addToast = useSetAtom(addToastAtom);
+    const form = useForm<FormValues>();
+
+    const onSubmit: SubmitHandler<FormValues> = async ({ id }) => {
+        if (accountTokens.some((token) => token.id === id)) {
+            addToast(t('duplicateId'));
+            return;
+        }
+        const tokenUrl = await getTokenUrl(client, id || '', contractDetails);
+        const meta = await getTokenMetadata(tokenUrl);
+        setAccountTokens((tokens) => [...tokens, { id, metadata: meta, metadataLink: tokenUrl }]);
+    };
+
+    return (
+        <div className="tokens__add__container">
+            <UncontrolledInput
+                readOnly
+                className="tokens__add__input"
+                label={t('contractIndex')}
+                value={contractDetails.index.toString()}
+            />
+            <UncontrolledInput
+                readOnly
+                className="tokens__add__input"
+                label={t('contractName')}
+                value={contractDetails.contractName}
+            />
+            <Form formMethods={form} className="tokens__add__add-token" onSubmit={onSubmit}>
+                {(f) => (
+                    <>
+                        <Input register={f.register} label={t('tokenId')} name="id" />
+                        <Submit>{t('addToken')}</Submit>
+                    </>
+                )}
+            </Form>
+            {accountTokens.map((token) => (
```

Comment:
I think it needs some some of header above the list.. What do you think?

```suggestion
            <h3>Tokens:</h3>
            {accountTokens.map((token) => (
```

Reply:
I changed it in another way :thinking:

Reply:
Looks good!